sdk/keyvault/keyvault/src/az_keyvault_client.c

```diff
@@ -79,7 +79,7 @@ AZ_NODISCARD az_result az_keyvault_keys_client_init(
             {
               ._internal = {
                 .process = az_http_pipeline_policy_apiversion,
-                .p_options= &self->_internal.options._internal.api_version,
+                .p_options= &options->_internal.api_version,
```

Comment:
No, this isn't right. The code was right before. Line 74 copies the *options by-value into the client's options. The lines you changes are setting p_options to the ADDRESS of the option structs. The ADDRESS is correct whether the data is in there yet or not. However, I think line 74 will execute first but it doesn't really matter.

We could add comments so that no one reading the code thinks that we're initializing to uninitialized data. I think all our tests worked correctly.

Reply:
We don't have sufficient testing on this space.  The POCs were hitting access violations b/c the assignments were pointing to random memory